spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn
+
+import breeze.linalg.dim
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.transform.vision.image.label.roi.RoiLabel
+import com.intel.analytics.bigdl.transform.vision.image.util.BboxUtil
+import com.intel.analytics.bigdl.utils.RandomGenerator._
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.collection.mutable.ArrayBuffer
+
+class BoxHead(
+  val inChannels: Int,
+  val resolution: Int,
+  val scales: Array[Float],
+  val samplingRatio: Float,
+  val scoreThresh: Float,
+  val nmsThresh: Float,
+  val maxPerImage: Int,
+  val outputSize: Int,
+  val numClasses: Int
+  )(implicit ev: TensorNumeric[Float])
+  extends BaseModule[Float] {
+
+  override def buildModel(): Module[Float] = {
+    val featureExtractor = this.featureExtractor(
+      inChannels, resolution, scales, samplingRatio.toInt, outputSize)
```

Comment:
why toInt ? can we just make the type as Int instead of Float ?

Reply:
ok, good idea. changed